File: js/1bitOr2bitString.js
var isOneBitCharacter = function(bits) {
	count = bits.length
    if(bits[count - 1] != 0)
        return false

    var find = false
	var i = count - 2
	while(i > 0){
		if (bits[i] == 0){
            find = true
			break
		}
		i--
	}
    if (find)
        i++
	return (count-1 - i) % 2 == 0
};

console.log(isOneBitCharacter([1,1,0]))

File: python/1bitOr2bitString.py
def isEndWith0(bits):
    count = len(bits)
    if bits[count-1] != 0:
        return False
    if count == 1:
        return True
    r = True
    i = count - 2
    while(i >= 0):
        if bits[i] == 0:
            break
        i-=1
        r = not r
    return r

def main():
    print(isEndWith0([1,1,0]))

if __name__ == '__main__':
    main()